ddtrace/internal/telemetry/writer.py

```diff
@@ -252,10 +261,15 @@ def _app_started_event(self):
             # app-started events should only be sent by the main process
             return
         #  List of configurations to be collected
+
         self.add_configurations(
             [
-                ("data_streams_enabled", config._data_streams_enabled, "unknown"),
-                ("appsec_enabled", config._appsec_enabled, "unknown"),
+                (TELEMETRY_TRACING_ENABLED, config._tracing_enabled, "unknown"),
+                (TELEMETRY_DSM_ENABLED, config._data_streams_enabled, "unknown"),
+                (TELEMETRY_ASM_ENABLED, config._appsec_enabled, "unknown"),
+                (TELEMETRY_PROFILING_ENABLED, profiling_config.enabled, "unknown"),
+                (TELEMETRY_DYNAMIC_INSTRUMENTATION_ENABLED, DynamicInstrumentationConfig().enabled, "unknown"),
+                (TELEMETRY_EXCEPTION_DEBUGGING_ENABLED, ExceptionDebuggingConfig().enabled, "unknown"),
```

Comment:
Some of these services support a programmatic API for enablement. If we only restrict to the env var configuration we might end up reporting enabled services as not enabled. Not sure if this issue would be in scope w.r.t. this PR though, mentioning just in case.

Reply:
Yup that's the big concern here and why I initially removed dynamic instrumentation support from this PR. 

We'll need to resolve the circular import before tracking the programatic api.